Fluid/Filters/MiscFilters.cs

```diff
@@ -411,34 +414,60 @@ void ForStrf(DateTimeOffset value, string format, StringBuilder result)
-                            case 'l': result.Append(value.ToString("%h", context.CultureInfo).PadLeft(2, ' ')); break;
-                            case 'L': result.Append(value.Millisecond.ToString(context.CultureInfo).PadLeft(3, '0')); break;
-                            case 'm':
+                            case 'I':
                                 {
-                                    var month = value.Month.ToString(context.CultureInfo);
-                                    if (useSpaceForPaddingFlag)
-                                    {
-                                        result.Append(month.PadLeft(2, ' '));
-                                    }
-                                    else if (removeLeadingZerosFlag)
+                                    var hour = value.Hour switch
                                     {
-                                        result.Append(month);
-                                    }
-                                    else
+                                        0 => 12,
+                                        <= 12 => value.Hour,
+                                        _ => value.Hour - 12
+                                    };
+
+                                    result.Append(Format(hour, 2));
+                                    break;
+                                }
+                            case 'j': result.Append(Format(value.DayOfYear, 3)); break;
+                            case 'k': result.Append(value.Hour); break;
+                            case 'l':
+                                {
+                                    useSpaceForPaddingFlag = true;
+                                    var hour = value.Hour switch
                                     {
-                                        result.Append(month.PadLeft(2, '0'));
-                                    }
+                                        0 => 12,
+                                        <= 12 => value.Hour,
+                                        _ => value.Hour - 12
+                                    };
+
+                                    result.Append(Format(hour, 2));
                                     break;
                                 }
+                            case 'L': result.Append(Format(value.Millisecond, 3)); break;
+                            case 'm': result.Append(Format(value.Month, 2)); break;
                             case 'M':
-                                result.Append(value.Minute.ToString(context.CultureInfo).PadLeft(2, '0'));
+                                result.Append(Format(value.Minute, 2));
                                 break;
+                            case 'n': result.Append(new String('\n', width ?? 1)); break;
+                            case 'N': var v = value.Millisecond.ToString(context.CultureInfo); width ??= 9;  result.Append(v.Length >= width ? v.Substring(0, width.Value) : v.PadRight(width.Value, '0')); break;
```

Comment:
I was interested to learn how this was implemented.  On this line, Millisecond is being used, however DateTime/Offset support down to 100ns precision via ticks.

Could you perhaps use something like:

```csharp
var v = (value.Ticks % 10000000).ToString(context.CultureInfo); // get part-second from ticks in 100ns precision.
width ?? = 9;
result.Append(v.Length >= width ? v.Substring(0, width.Value) : v.PadRight(width.Value, '0'));
break;
```

Reply:
Thanks for the suggestion. The nanoseconds are parsed to the 7th digit but that's already an improvement. Also note that net7.0 introduced `.Microseconds` and `.Nanoseconds` properties, however they don't represent what I would have expected.